### ISSUE TITLE
configs override shader presets

### ISSUE THREAD

**Issue by pepedopolous** [libretro/RetroArch#3514]
I tried the new shader presets per core but this does not work if you already have a config override for that core (even if there is absolutely nothing shader-related in the override).

To make it work I had to specify the shader preset in the config override e.g.

~/.config/retroarch/config/FB Alpha/FB Alpha.cfg
input_overlay = "foo.cfg"
video_shader = "/.config/shaders/presets/FB Alpha/FB Alpha.slangp

I think it would be better if the shader preset per core or per game was applied unless there is a shader specified in the config override. So the config override is simply used for overlays, custom resolutions, aspect ratio etc. but not shaders as there is a GUI system already in place for that.

**Comment by fr500** [comment]
it's probably just a matter of doing presets after overrides, I'll take a look later.

**Comment by fr500** [comment]
It worked fine for me here with CG.
Maybe I have a problem with slangp I saw that mentioned at least once before, can you try?

**Comment by pepedopolous** [comment]
I just deleted the FB Alpha config override and shader preset. Then I made a new shader preset (I loaded crt-easymode-halation.slangp, tweaked the geom settings, and saved the preset for this core.)

However, it doesn't load at all when I re-start FBA. I have to manually load and apply the preset to get it to work...

**Comment by fr500** [comment]
can you try with CG or GLSL?

**Comment by pepedopolous** [comment]
I will but I can't tonight... About your earlier comment... I did do the presets after the overrides as I originally had the overrides in place to specify the overlay, res, and shader choice. I simply deleted the line for the shader, loaded the game, and then tried saving a preset.

**Comment by fr500** [comment]
No I didn't mean what you do, I mean in the codepath but it's correct.

**Comment by Ryunam** [comment]
Possibly related to [https://github.com/libretro/RetroArch/issues/3504](url)?

The issue I had described in the aforementioned post happened more or less in the same way. I had created a Config Override for bsnes-mercury-balanced, then proceeded to save a certain Shader Preset (in my case, it occurred with the CG shader "pixellate") but it would refuse to apply it automatically upon loading content.

I fixed it in a different way compared to pepdopolous, that is by opening "Load Shader Preset" within the Quick Menu and manually specifying once again the default preset I wanted to load with that core.

**Comment by fr500** [comment]
It's still working fine for me.
I just tried with CG and GLSL, no vulkan on this PC.

I set one preset on a core with no overrides and another preset on a core WITH overrides and cycled between 4 different games and the shader always applied as expected.

**Comment by pepedopolous** [comment]
Hmm. I just tried using OpenGL and Cg (crt-geom-crt).  It doesn't happen in exactly the same way as Vulkan/glslang. The shader is looks like it is loaded in the shader menu but it isn't actually. This is obvious as the screen geometry is unchanged and no parameters are listed for the shader. However, I can load the shader manually (either the orginal crt-geom-cgp or the core preset (/shaders/presets/FB Alpha/FB Alpha.cgp). I also tried with a glsl preset.

http://pastebin.com/2zmEyvGh
